phpunit/functional/TicketTest.php

```diff
@@ -7523,6 +7523,36 @@ protected function timelineItemsProvider(): iterable
                 ],
             ];
         }
+
+        $profile_right->getFromDBByCrit([
+            'profiles_id' => $tprofile_id,
+            'name'        => 'task',
+        ]);
+
+        $this->updateItem(
+            \ProfileRight::class,
+            $profile_right->getID(),
+            [
+                'rights' => \CommonITILTask::SEEPUBLIC,
+            ]
+        );
+
+        // tech will only see own private tasks and all private followups
+        yield [
+            'login'              => 'tech',
+            'pass'               => 'tech',
+            'ticket_id'          => $ticket->getID(),
+            'options'            => [],
+            'expected_followups' => [
+                'private followup of normal user',
+                'private followup of tech user',
+                'public followup',
+            ],
+            'expected_tasks'     => [
+                'private task of tech user',
+                'public task',
+            ],
+        ];
```

Comment:
This new case does not validate the filtering on the `users_id_tech` field, and matches the exact same result as the case provided on lines 7436->7451. Instead of adding this new case, you should add a new TicketTask (e.g. `private task assigned to tech user`) and update existing cases.